libraries/core_libs/network/src/tarcap/shared_states/pbft_syncing_state.cpp

```diff
@@ -65,4 +65,32 @@ bool PbftSyncingState::isPbftSyncing() {
   return pbft_syncing_;
 }
 
+bool PbftSyncingState::updatePeerAskingPeriod(const dev::p2p::NodeID& peer_id, uint64_t period) {
+  auto now = std::chrono::steady_clock::now();
+
+  UpgradableLock lock(asking_period_mutex_);
+  if (asking_period_peers_table_.contains(peer_id) && asking_period_peers_table_[peer_id].first >= period &&
+      std::chrono::duration_cast<std::chrono::milliseconds>(now - asking_period_peers_table_[peer_id].second) <=
+          kPeerAskingPeriodTimeout) {
+    return false;
+  }
+
+  UpgradeLock locked(lock);
+  asking_period_peers_table_[peer_id] = std::make_pair(period, now);
+
+  return true;
```

Comment:
you should be cleaning this table on disconnect otherwise it will have a lot of old data. Or maybe when the syncing is finished

Reply:
You are right. Nodes will not connect again after mark malicious. Erase the malicious peers from table